packages/frontend/src/utils/assetUtils.ts

```diff
@@ -0,0 +1,112 @@
+import { AssetDetails } from '@explorer/shared'
+import { AssetHash, AssetId } from '@explorer/types'
+
+export interface AssetInfo {
+  name: string
+  symbol: string
+  isUnknownHash?: boolean
+  imageUrl: string
+}
+
+export function assetToInfo(
+  asset: AssetId | AssetHash,
+  assetDetails?: AssetDetails
+): AssetInfo {
+  return AssetId.check(asset)
+    ? assetIdToInfo(asset)
+    : assetHashToInfo(asset, assetDetails)
+}
+
+function assetIdToInfo(assetId: AssetId): AssetInfo {
+  const symbol = assetId.split('-')[0] ?? ''
+  const assetInfo = perpetualsInfo[symbol]
+
+  if (!assetInfo) {
+    return {
+      name: symbol,
+      symbol,
+      imageUrl: '/images/unknown-asset.svg',
+    }
+  }
+
+  return {
+    ...assetInfo,
+    imageUrl: getDydxImageUrl(assetInfo.symbol),
+  }
+}
+
+function assetHashToInfo(
+  assetHash: AssetHash,
+  assetDetails?: AssetDetails
+): AssetInfo {
+  if (!assetDetails) {
+    return {
+      name: 'Unknown',
+      symbol: assetHash.toString(),
```

Comment:
It's not a good idea to use AssetHash converted to string as symbol here. I would use 'unknown' or something like that instead.

Reply:
That is how it is designed
![image](https://user-images.githubusercontent.com/93620601/219584173-c6ec643a-a944-4536-97ce-b43335f42b69.png)

I think only thing missing is shortening of an address